gui/src/installer/step/node/mod.rs

```diff
@@ -0,0 +1,247 @@
+pub mod bitcoind;
+pub mod electrum;
+
+use crate::{
+    hw::HardwareWallets,
+    installer::{
+        context::Context,
+        message::{self, Message},
+        step::{
+            node::{bitcoind::DefineBitcoind, electrum::DefineElectrum},
+            Step,
+        },
+        view, Error,
+    },
+    node::NodeType,
+};
+
+use iced::Command;
+use liana_ui::widget::Element;
+
+#[derive(Clone)]
+pub enum NodeDefinition {
+    Bitcoind(DefineBitcoind),
+    Electrum(DefineElectrum),
+}
+
+impl NodeDefinition {
+    fn new(node_type: &NodeType) -> Self {
+        match node_type {
+            NodeType::Bitcoind => NodeDefinition::Bitcoind(DefineBitcoind::new()),
+            NodeType::Electrum => NodeDefinition::Electrum(DefineElectrum::new()),
+        }
+    }
+
+    fn node_type(&self) -> NodeType {
+        match self {
+            NodeDefinition::Bitcoind(_) => NodeType::Bitcoind,
+            NodeDefinition::Electrum(_) => NodeType::Electrum,
+        }
+    }
+
+    fn apply(&mut self, ctx: &mut Context) -> bool {
+        match self {
+            NodeDefinition::Bitcoind(def) => def.apply(ctx),
+            NodeDefinition::Electrum(def) => def.apply(ctx),
+        }
+    }
+
+    fn can_try_ping(&self) -> bool {
+        match self {
+            NodeDefinition::Bitcoind(def) => def.can_try_ping(),
+            NodeDefinition::Electrum(def) => def.can_try_ping(),
+        }
+    }
+
+    fn load_context(&mut self, ctx: &Context) {
+        match self {
+            NodeDefinition::Bitcoind(def) => def.load_context(ctx),
+            NodeDefinition::Electrum(def) => def.load_context(ctx),
+        }
+    }
+
+    fn update(&mut self, message: message::DefineNode) -> Command<Message> {
+        match self {
+            NodeDefinition::Bitcoind(def) => def.update(message),
+            NodeDefinition::Electrum(def) => def.update(message),
+        }
+    }
+
+    fn view(&self) -> Element<Message> {
+        match self {
+            NodeDefinition::Bitcoind(def) => def.view(),
+            NodeDefinition::Electrum(def) => def.view(),
+        }
+    }
+
+    fn ping(&self) -> Result<(), Error> {
+        match self {
+            NodeDefinition::Bitcoind(def) => def.ping(),
+            NodeDefinition::Electrum(def) => def.ping(),
+        }
+    }
+}
+
+pub struct Node {
+    definition: NodeDefinition,
+    is_running: Option<Result<(), Error>>,
+    waiting_for_ping_result: bool,
+}
+
+impl Node {
+    fn new(node_type: &NodeType) -> Self {
```

Comment:
`NodeType` implement `Copy`
```suggestion
    fn new(node_type: NodeType) -> Self {
```